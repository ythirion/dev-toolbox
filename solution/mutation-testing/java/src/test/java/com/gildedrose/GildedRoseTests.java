package com.gildedrose;

import org.junit.jupiter.api.Test;

import static org.approvaltests.combinations.CombinationApprovals.verifyAllCombinations;

class GildedRoseTests {
    @Test
    void updateQuality() {
        verifyAllCombinations(
                this::updateQualityFor,
                new String[]{"a common item",
                        "Aged Brie",
                        "Backstage passes to a TAFKAL80ETC concert",
                        "Sulfuras, Hand of Ragnaros"},
                new Integer[]{-100, -1, 0, 1, 5, 6, 7, 10, 11, 12},
                new Integer[]{0, 1, 49, 50, 51}
        );
    }

    private String updateQualityFor(String name, int sellIn, int quality) {
        var items = new Item[]{new Item(name, sellIn, quality)};
        var gildedRose = new GildedRose(items);
        gildedRose.updateQuality();

        return gildedRose.items[0].toString();
    }
}