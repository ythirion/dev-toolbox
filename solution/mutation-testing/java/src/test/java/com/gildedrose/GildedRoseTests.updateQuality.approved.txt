[a common item, -100, 0] => a common item, -101, 0 
[a common item, -100, 1] => a common item, -101, 0 
[a common item, -100, 49] => a common item, -101, 47 
[a common item, -100, 50] => a common item, -101, 48 
[a common item, -100, 51] => a common item, -101, 49 
[a common item, -1, 0] => a common item, -2, 0 
[a common item, -1, 1] => a common item, -2, 0 
[a common item, -1, 49] => a common item, -2, 47 
[a common item, -1, 50] => a common item, -2, 48 
[a common item, -1, 51] => a common item, -2, 49 
[a common item, 0, 0] => a common item, -1, 0 
[a common item, 0, 1] => a common item, -1, 0 
[a common item, 0, 49] => a common item, -1, 47 
[a common item, 0, 50] => a common item, -1, 48 
[a common item, 0, 51] => a common item, -1, 49 
[a common item, 1, 0] => a common item, 0, 0 
[a common item, 1, 1] => a common item, 0, 0 
[a common item, 1, 49] => a common item, 0, 48 
[a common item, 1, 50] => a common item, 0, 49 
[a common item, 1, 51] => a common item, 0, 50 
[a common item, 5, 0] => a common item, 4, 0 
[a common item, 5, 1] => a common item, 4, 0 
[a common item, 5, 49] => a common item, 4, 48 
[a common item, 5, 50] => a common item, 4, 49 
[a common item, 5, 51] => a common item, 4, 50 
[a common item, 6, 0] => a common item, 5, 0 
[a common item, 6, 1] => a common item, 5, 0 
[a common item, 6, 49] => a common item, 5, 48 
[a common item, 6, 50] => a common item, 5, 49 
[a common item, 6, 51] => a common item, 5, 50 
[a common item, 7, 0] => a common item, 6, 0 
[a common item, 7, 1] => a common item, 6, 0 
[a common item, 7, 49] => a common item, 6, 48 
[a common item, 7, 50] => a common item, 6, 49 
[a common item, 7, 51] => a common item, 6, 50 
[a common item, 10, 0] => a common item, 9, 0 
[a common item, 10, 1] => a common item, 9, 0 
[a common item, 10, 49] => a common item, 9, 48 
[a common item, 10, 50] => a common item, 9, 49 
[a common item, 10, 51] => a common item, 9, 50 
[a common item, 11, 0] => a common item, 10, 0 
[a common item, 11, 1] => a common item, 10, 0 
[a common item, 11, 49] => a common item, 10, 48 
[a common item, 11, 50] => a common item, 10, 49 
[a common item, 11, 51] => a common item, 10, 50 
[a common item, 12, 0] => a common item, 11, 0 
[a common item, 12, 1] => a common item, 11, 0 
[a common item, 12, 49] => a common item, 11, 48 
[a common item, 12, 50] => a common item, 11, 49 
[a common item, 12, 51] => a common item, 11, 50 
[Aged Brie, -100, 0] => Aged Brie, -101, 2 
[Aged Brie, -100, 1] => Aged Brie, -101, 3 
[Aged Brie, -100, 49] => Aged Brie, -101, 50 
[Aged Brie, -100, 50] => Aged Brie, -101, 50 
[Aged Brie, -100, 51] => Aged Brie, -101, 51 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, -1, 51] => Aged Brie, -2, 51 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 0, 51] => Aged Brie, -1, 51 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 1] => Aged Brie, 0, 2 
[Aged Brie, 1, 49] => Aged Brie, 0, 50 
[Aged Brie, 1, 50] => Aged Brie, 0, 50 
[Aged Brie, 1, 51] => Aged Brie, 0, 51 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 1] => Aged Brie, 4, 2 
[Aged Brie, 5, 49] => Aged Brie, 4, 50 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 5, 51] => Aged Brie, 4, 51 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 1] => Aged Brie, 5, 2 
[Aged Brie, 6, 49] => Aged Brie, 5, 50 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 6, 51] => Aged Brie, 5, 51 
[Aged Brie, 7, 0] => Aged Brie, 6, 1 
[Aged Brie, 7, 1] => Aged Brie, 6, 2 
[Aged Brie, 7, 49] => Aged Brie, 6, 50 
[Aged Brie, 7, 50] => Aged Brie, 6, 50 
[Aged Brie, 7, 51] => Aged Brie, 6, 51 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 1] => Aged Brie, 9, 2 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 10, 51] => Aged Brie, 9, 51 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Aged Brie, 11, 51] => Aged Brie, 10, 51 
[Aged Brie, 12, 0] => Aged Brie, 11, 1 
[Aged Brie, 12, 1] => Aged Brie, 11, 2 
[Aged Brie, 12, 49] => Aged Brie, 11, 50 
[Aged Brie, 12, 50] => Aged Brie, 11, 50 
[Aged Brie, 12, 51] => Aged Brie, 11, 51 
[Backstage passes to a TAFKAL80ETC concert, -100, 0] => Backstage passes to a TAFKAL80ETC concert, -101, 0 
[Backstage passes to a TAFKAL80ETC concert, -100, 1] => Backstage passes to a TAFKAL80ETC concert, -101, 0 
[Backstage passes to a TAFKAL80ETC concert, -100, 49] => Backstage passes to a TAFKAL80ETC concert, -101, 0 
[Backstage passes to a TAFKAL80ETC concert, -100, 50] => Backstage passes to a TAFKAL80ETC concert, -101, 0 
[Backstage passes to a TAFKAL80ETC concert, -100, 51] => Backstage passes to a TAFKAL80ETC concert, -101, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 51] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 51] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 1] => Backstage passes to a TAFKAL80ETC concert, 0, 4 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 51] => Backstage passes to a TAFKAL80ETC concert, 0, 51 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 51] => Backstage passes to a TAFKAL80ETC concert, 4, 51 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] => Backstage passes to a TAFKAL80ETC concert, 5, 3 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 51] => Backstage passes to a TAFKAL80ETC concert, 5, 51 
[Backstage passes to a TAFKAL80ETC concert, 7, 0] => Backstage passes to a TAFKAL80ETC concert, 6, 2 
[Backstage passes to a TAFKAL80ETC concert, 7, 1] => Backstage passes to a TAFKAL80ETC concert, 6, 3 
[Backstage passes to a TAFKAL80ETC concert, 7, 49] => Backstage passes to a TAFKAL80ETC concert, 6, 50 
[Backstage passes to a TAFKAL80ETC concert, 7, 50] => Backstage passes to a TAFKAL80ETC concert, 6, 50 
[Backstage passes to a TAFKAL80ETC concert, 7, 51] => Backstage passes to a TAFKAL80ETC concert, 6, 51 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 9, 3 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 51] => Backstage passes to a TAFKAL80ETC concert, 9, 51 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 51] => Backstage passes to a TAFKAL80ETC concert, 10, 51 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Backstage passes to a TAFKAL80ETC concert, 11, 1 
[Backstage passes to a TAFKAL80ETC concert, 12, 1] => Backstage passes to a TAFKAL80ETC concert, 11, 2 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 51] => Backstage passes to a TAFKAL80ETC concert, 11, 51 
[Sulfuras, Hand of Ragnaros, -100, 0] => Sulfuras, Hand of Ragnaros, -100, 0 
[Sulfuras, Hand of Ragnaros, -100, 1] => Sulfuras, Hand of Ragnaros, -100, 1 
[Sulfuras, Hand of Ragnaros, -100, 49] => Sulfuras, Hand of Ragnaros, -100, 49 
[Sulfuras, Hand of Ragnaros, -100, 50] => Sulfuras, Hand of Ragnaros, -100, 50 
[Sulfuras, Hand of Ragnaros, -100, 51] => Sulfuras, Hand of Ragnaros, -100, 51 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, -1, 51] => Sulfuras, Hand of Ragnaros, -1, 51 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 51] => Sulfuras, Hand of Ragnaros, 0, 51 
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros, 1, 0 
[Sulfuras, Hand of Ragnaros, 1, 1] => Sulfuras, Hand of Ragnaros, 1, 1 
[Sulfuras, Hand of Ragnaros, 1, 49] => Sulfuras, Hand of Ragnaros, 1, 49 
[Sulfuras, Hand of Ragnaros, 1, 50] => Sulfuras, Hand of Ragnaros, 1, 50 
[Sulfuras, Hand of Ragnaros, 1, 51] => Sulfuras, Hand of Ragnaros, 1, 51 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 1] => Sulfuras, Hand of Ragnaros, 5, 1 
[Sulfuras, Hand of Ragnaros, 5, 49] => Sulfuras, Hand of Ragnaros, 5, 49 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[Sulfuras, Hand of Ragnaros, 5, 51] => Sulfuras, Hand of Ragnaros, 5, 51 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[Sulfuras, Hand of Ragnaros, 6, 1] => Sulfuras, Hand of Ragnaros, 6, 1 
[Sulfuras, Hand of Ragnaros, 6, 49] => Sulfuras, Hand of Ragnaros, 6, 49 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[Sulfuras, Hand of Ragnaros, 6, 51] => Sulfuras, Hand of Ragnaros, 6, 51 
[Sulfuras, Hand of Ragnaros, 7, 0] => Sulfuras, Hand of Ragnaros, 7, 0 
[Sulfuras, Hand of Ragnaros, 7, 1] => Sulfuras, Hand of Ragnaros, 7, 1 
[Sulfuras, Hand of Ragnaros, 7, 49] => Sulfuras, Hand of Ragnaros, 7, 49 
[Sulfuras, Hand of Ragnaros, 7, 50] => Sulfuras, Hand of Ragnaros, 7, 50 
[Sulfuras, Hand of Ragnaros, 7, 51] => Sulfuras, Hand of Ragnaros, 7, 51 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 1] => Sulfuras, Hand of Ragnaros, 10, 1 
[Sulfuras, Hand of Ragnaros, 10, 49] => Sulfuras, Hand of Ragnaros, 10, 49 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 10, 51] => Sulfuras, Hand of Ragnaros, 10, 51 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 1 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
[Sulfuras, Hand of Ragnaros, 11, 51] => Sulfuras, Hand of Ragnaros, 11, 51 
[Sulfuras, Hand of Ragnaros, 12, 0] => Sulfuras, Hand of Ragnaros, 12, 0 
[Sulfuras, Hand of Ragnaros, 12, 1] => Sulfuras, Hand of Ragnaros, 12, 1 
[Sulfuras, Hand of Ragnaros, 12, 49] => Sulfuras, Hand of Ragnaros, 12, 49 
[Sulfuras, Hand of Ragnaros, 12, 50] => Sulfuras, Hand of Ragnaros, 12, 50 
[Sulfuras, Hand of Ragnaros, 12, 51] => Sulfuras, Hand of Ragnaros, 12, 51 
